src/som/interpreter/actors/SPromise.java

```diff
@@ -230,6 +274,72 @@ final Object getValueUnsync() {
     return value;
   }
 
+  public int getNumChainedPromises() {
```

Comment:
Everything here in this changed block is inherently racy.

Especially, if we do not do a stop the world, then this is going to break if we do not synchronize this properly somehow.

Reply:
Probably would be best to perform all the gets in one synchronised block in the PromiseSerializationNode, and add Unsynced to the method names.